lib/kb_ObjectInfo/kb_ObjectInfoImpl.py

```diff
@@ -518,21 +541,71 @@ def genomeset_report(self, ctx, params):
         multi_fasta = []
         rpt_string = ''
         
+#       Always do the list of genomes - short and ensures that there is an HTML link
+        gsr = CreateMultiGenomeReport(self.config)
+        rpt_list = gsr.readGenomeSet(genome_name, genomeset_data)
+        htmltable = self.make_HTML(rpt_list,'col_header')
+        html_report_path = os.path.join(self.scratch, 'genomeset_comparison.html')
+        html_report_txt = open(html_report_path, "w")
+        html_report_txt.write("<h1>GENOME COMPARISON</h1>")
+        html_report_txt.write(htmltable)
+        html_report_txt.close()
+        
         if params['showGenomes']:
-            gsr = CreateMultiGenomeReport(self.config)
-            rpt_list = gsr.readGenomeSet(genome_name, genomeset_data)
             rpt_string += self.write_to_file(rpt_list,'genomeset_tab_file.tsv',"\t")
             self.write_to_file(rpt_list,'genomeset_cvs_file.csv',",")
-            htmltable = self.make_HTML(rpt_list,'col_header')
-            
-            html_report_path = os.path.join(self.scratch, 'genomeset_comparison.html')
+
+        if params['listCoding']:
+            cf = CreateFeatureLists(self.config)
+            myGS = genomeset_data['elements']
+            rpt_list = []
+            for ele in myGS:
+                genome = self.dfu.get_objects({'object_refs': [myGS[ele]['ref']]})
+                genome_data = genome['data'][0]['data']
+
+                rpt_list += cf.delimitedTable(genome_data, 'features')
+                
+            if rpt_list:
+                rpt_string += self.write_to_file(rpt_list,'genome_tab_file.tsv',"\t")
+                self.write_to_file(rpt_list,'genome_csv_file.csv',",")
+                
+            html_report_path = os.path.join(self.scratch, 'genome_features.html')
             html_report_txt = open(html_report_path, "w")
-            html_report_txt.write("<h1>GENOME COMPARISON</h1>")
+            htmltable = self.make_HTML(rpt_list,'col_header')
+            html_report_txt.write("<h1>PROTEIN CODING FEATURES</h1>")
             html_report_txt.write(htmltable)
             html_report_txt.close()
-        
+                
+        if params['listGFF']:
+            myGS = genomeset_data['elements']
+    
+            for ele in myGS:
+                gff_return = self.gfu.genome_to_gff({'genome_ref':myGS[ele]['ref'],'is_gtf':0,'target_dir':self.scratch})
+                rpt_string += "Created " + gff_return['file_path'] + "\n"
+             
+        if params['listGBK']:
+            myGS = genomeset_data['elements']
+    
+            for ele in myGS:
```

Comment:
This pair of shorthand names is an example of one that might be hard to understand in isolation, though the line above would help someone figure it out. But who needs the extra mental load of having to translate mentally every time it is read?